platform/core/src/types/Consumer.ts

```diff
@@ -0,0 +1,3 @@
+import Params from './ParamsType';
+type Consumer = (props: Params) => void;
+export default Consumer;
```

Comment:
What is Consumer? I don't think we need this level of typings, but if you want, can we maybe call it like `Callback`?

Reply:
Also I could not comment on `StudyMetadata` but that file is empty and should be removed

Reply:
That was a deleted file.

Add documentation to Consumer - that is a standard definition/actor type, and is useful to have a name for rather than declaring it every time it gets used - they are pretty common.

Reply:
Params are `Record<string, unknown` and `Consumer` is getting `Params` as its function argument. The reason I say we don't need this is that there can be multiple (if not infinite) combination of all of this 

Params that is not just one parameter, a `Record<string, unkonwn` and another optional parameter, then based on this you probably will say `myFunction(params: Params, anotherParam: AnotherType)`. We can't assume the parameters of a function are just in type of `Record<string, unkonwn` neither we can assume that a "consumer" is always taking one argument and is a void in return. What about another consumer which returns something? 

In summary, calling these Params and Consumer is a very general term which adds a lot to ambiguity, if they are event handler's parameters and consumer, then let's call them EventHandlerParam, and EventHandlerFunction or Consumer or whatever.